protocol/bc/outputcommitment.go

```diff
@@ -0,0 +1,66 @@
+package bc
+
+import (
+	"fmt"
+	"io"
+
+	"chain/encoding/blockchain"
+	"chain/encoding/bufpool"
+	"chain/errors"
+)
+
+// OutputCommitment contains the commitment data for a transaction
+// output (which also appears in the spend input of that output).
+type OutputCommitment struct {
+	AssetAmount
+	VMVersion      uint64
+	ControlProgram []byte
+}
+
+func (oc *OutputCommitment) writeTo(w io.Writer, assetVersion uint64) (err error) {
+	b := bufpool.Get()
```

Comment:
The entire body of this function, except for the `return` at the end, should be replaced with a call to `WriteExtensibleString` (analogous to the `ReadExtensibleString` call in `readFrom`, below).

The callback to `WriteExtensibleString` should still wrap its nested writes within an `if assetVersion == 1` test.